core/chain-configs/src/genesis_config.rs

```diff
@@ -337,6 +457,33 @@ impl Genesis {
         hasher.process_genesis(self);
         hasher.finalize()
     }
+
+    pub fn stream_records_with_callback(&self, callback: impl FnMut(StateRecord)) {
```

Comment:
Let's inline the function into the caller? At the moment, the public API looks like this:

```
    pub fn stream_records_with_callback(&self, callback: impl FnMut(StateRecord))
    pub fn process_records(&self, mut callback: impl FnMut(&StateRecord))
```

It's not really clear what's the differences between the two fns. Making one private (or maybe just getting rid of it altogether) makes the API much clearer.

Reply:
Made stream_records_with_callback private